src/test/java/com/virtuslab/using_directives/parser/ParserUnitTest.java

```diff
@@ -0,0 +1,164 @@
+package com.virtuslab.using_directives.parser;
+
+import static com.virtuslab.using_directives.DirectiveAssertions.*;
+import static com.virtuslab.using_directives.TestUtils.*;
+
+import com.virtuslab.using_directives.custom.model.UsingDirectives;
+import com.virtuslab.using_directives.reporter.PersistentReporter;
+import org.junit.jupiter.api.Test;
+
+public class ParserUnitTest {
+
+  @Test
+  public void testNestedAndValuesOnOneLevel() {
+    String input = joinLines("using", "   keyA \"valueA\"", "   keyB: ", "      keyC \"valueC\"");
+    UsingDirectives parsedDirective = testCode(2, input);
+    assertSamePaths(parsedDirective, "keyA", "keyB.keyC");
+  }
+
+  @Test
+  public void testEmptyListElem() {
+    String input = "using keyA 0, , 2";
+    UsingDirectives parsedDirective = testCode(1, input);
+    assertValueListSize(parsedDirective, "keyA", 3);
+  }
+
+  @Test
+  public void testBooleanLiteral() {
+    String input = "using keyA false";
+    UsingDirectives parsedDirective = testCode(1, input);
+    assertValueAtPath(parsedDirective, "keyA", "false");
+  }
+
+  @Test
+  public void testFailMissingBrace() {
+    PersistentReporter reporter = reporterAfterParsing("using keyA {", "   keyB 18", "x");
+    assertDiagnostic(reporter, 2, 0, "Expected closing region token but found identifier");
+  }
+
+  @Test
+  public void testFailExpectingIndentOrBrace() {
+    PersistentReporter reporter = reporterAfterParsing("using (keyB 2)");
+    assertDiagnostic(reporter, 0, 6, "Expected indent or braces but found", "(");
```

Comment:
I do not thing that we use parentheses anywhere in our syntax so I think we can say that are not supported.